File: api/code-samples/javascript/klay/transaction/estimateGas.js
const OpenSdk = require("opensdk-javascript");

(() => {
    const sdk = new OpenSdk(new Caver.ApiClient("https://api.baobab.klaytn.net:8651"));

    const callObject = {
        "from": "0x3f71029af4e252b25b9ab999f77182f0cd3bc085",
        "to": "0x87ac99835e67168d4f9a40580f8f5c33550ba88b",
        "gas": "0x100000",
        "gasPrice": "0x5d21dba00",
        "value": "0x0",
        "input": "0x8ada066e"
    }
    sdk.klay.estimateGas(callObject, {}, (err, data, response) => {
        console.log(data);
    });
}
)()

File: api/code-samples/python/eth/transaction/estimateGas.py
from opensdk.sdk import OpenSDK

host = "https://api.baobab.klaytn.net:8651"

callObject = {
    "from": "0x51239f87c33e95e3bdb72e31d06b5306bcec81cc",
    "to": "0x87ac99835e67168d4f9a40580f8f5c33550ba88b",
    "gas": "0x100000",
    "gasPrice": "0x5d21dba00",
    "value": "0x0",
    "input": "0x8ada066e"
}

sdk = OpenSDK(host)
eth_response = sdk.eth.estimate_gas(callObject)

print(eth_response)
